src/components/User/User.tsx

```diff
@@ -0,0 +1,148 @@
+import React, { FC } from 'react';
+import { Input, Row, Button, Checkbox, Typography, Col, Form } from 'antd';
+
+import { userFormRules } from '../../constants/userConstants';
+
+import './styles.css';
+  
+type Props = {
+    firstName?: string;
+    lastName?: string;
+    email?: string;
+    password?: string;
+    newsletter?: boolean;
+};
+
+type DefaultProps = {
+    firstName: string;
+    lastName: string;
+    email: string;
+    password: string;
+    newsletter: boolean;
+};
+
+const defaultProps: DefaultProps = { 
+    firstName: '',
+    lastName: '',
+    email: '',
+    password: '',
+    newsletter: true
+};
+
+export interface UserFormValues {
+    firstName?: string;
+    lastName?: string;
+    email?: string;
+    password?: string;
+    newsletter?: boolean;
+};
+
+export interface UserFormRules {
+    firstName: Array<Object>;
+    lastName: Array<Object>;
+    email: Array<Object>;
+    password: Array<Object>;
+};
+
+const getInitialValues = (props: Props): UserFormValues => {
```

Comment:
UserFormValues and Props have the same interface but different types. This will make type error in strict mode

P.S. All fields are required

Reply:
Should I change UserFormValues interface so that all its fields are required?